sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c

```diff
@@ -197,7 +197,7 @@ ipv6_hdr_t *gnrc_ipv6_get_header(gnrc_pktsnip_t *pkt)
 {
     ipv6_hdr_t *hdr = NULL;
     gnrc_pktsnip_t *tmp = gnrc_pktsnip_search_type(pkt, GNRC_NETTYPE_IPV6);
-    if ((tmp != NULL) && (tmp->data != NULL) && ipv6_hdr_is(tmp->data)) {
+    if ((tmp != NULL) && (tmp->data != NULL) && ipv6_hdr_is(tmp->data) && tmp->size >= sizeof (ipv6_hdr_t)) {
```

Comment:
As mentioned in the top post of this review: I think it is a better idea to put this into an `assert()`. That needs to be then documented as a pre-condition for this function of course and would be an API change. But this way we can safe some bytes and as far as I can tell this function is only used within this PR and the unittests (though you wrote something about GNRC sock [which for me is [this](https://github.com/RIOT-OS/RIOT/tree/b4664a5bd307f1ba408223f360ee6a6c00ee1320/sys/net/gnrc/sock) I wasn't quite able to parse in your OP).

Reply:
Updated; all but `tmp != NULL` is now in asserts. (I'll happily change it to assert that there's an IPv6 header, but that felt like a legitimate result as opposed to the other checks).

My GNRC sock referece was about the change in gnrc_sock.c -- that originally does not use the `gnrc_ipv6_get_header` function but duplicates its functionality in `gnrc_sock_recv`, this patch makes it the first in-code user of `gnrc_ipv6_get_header`.

Reply:
> My GNRC sock referece was about the change in gnrc_sock.c -- that originally does not use the gnrc_ipv6_get_header function but duplicates its functionality in gnrc_sock_recv, this patch makes it the first in-code user of gnrc_ipv6_get_header.

I somehow was convinced that this PR edited `gnrc_ipv6.c` only...